packages/view-types/statistical-plots/src/ExpressionHistogramSubscriber.js

```diff
@@ -68,25 +90,46 @@ export function ExpressionHistogramSubscriber(props) {
   // generate the array of data points for the histogram.
   const data = useMemo(() => {
     if (firstGeneSelected && obsFeatureMatrix && expressionData) {
-      // Create new cellColors map based on the selected gene.
-      return Array.from(expressionData[0]).map((_, index) => {
-        const value = expressionData[0][index];
+      const newData = obsIndex.map((cellId, cellIndex) => {
+        const value = expressionData[0][cellIndex];
+        // Create new cellColors map based on the selected gene.
         const normValue = value * 100 / 255;
-        return { value: normValue, gene: firstGeneSelected };
+        const newItem = { value: normValue, gene: firstGeneSelected, cellId };
+        return newItem;
       });
+      setDataOnSelect(newData);
+      return newData;
     }
     if (obsFeatureMatrix) {
       const numGenes = featureIndex.length;
-      return obsIndex.map((cellId, cellIndex) => {
+      const newData = obsIndex.map((cellId, cellIndex) => {
         const values = obsFeatureMatrix.data
           .subarray(cellIndex * numGenes, (cellIndex + 1) * numGenes);
         const sumValue = sum(values) * 100 / 255;
-        return { value: sumValue, gene: null };
+        const newItem = { value: sumValue, gene: null, cellId };
+        return newItem;
       });
+      setDataOnSelect(newData);
+      return newData;
     }
     return null;
   }, [obsIndex, featureIndex, obsFeatureMatrix, firstGeneSelected, expressionData]);
 
+  const onSelect = useCallback((value) => {
+    const geneName = firstGeneSelected ? [firstGeneSelected, 'values'].join(' ') : 'transcript count';
+
+    const selectedCellIds = getObsInfoFromDataWithinRange(value, dataOnSelectRef.current);
+    setObsSelection(
+      selectedCellIds, additionalCellSets, cellSetColor,
+      setCellSetSelection, setAdditionalCellSets, setCellSetColor,
+      setCellColorEncoding,
+      'Selection ',
+      `: based on ${geneName} in range [${value[0].toFixed(1)}, ${value[1].toFixed(1)}] `,
+    );
+  }, [additionalCellSets, cellSetColor, data, setAdditionalCellSets,
+    setCellColorEncoding, setCellSetColor, setCellSetSelection,
```

Comment:
```suggestion
    setCellColorEncoding, setCellSetColor, setCellSetSelection, firstGeneSelected,
```

To fix linting warning